File: Arrays & Hashing/valid_anagram.cpp
#include <string>
#include <unordered_map>

class Solution {
public: 
    bool isAnagram(std::string s, std::string t)
    {
        if (t.size() != s.size()) return false;

        std::unordered_map<char, int> tmap;
        std::unordered_map<char, int> smap;

        for (int i = 0; i < s.size(); i++)
        {
            tmap[s[i]]++;
            smap[t[i]]++;
        }

        for (int i = 0; i < smap.size(); i++)
        {
            if (smap[i] != tmap[i]) return false;
        }
        return true;
    }



};

File: Arrays & Hashing/valid anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        
        hash_s, hash_m = {}, {}

        for i in range(len(s)):
            hash_s[s[i]] = 1 + hash_s.get(s[i], 0)
            hash_m[t[i]] = 1 + hash_m.get(t[i], 0)


        for c in hash_s:
            if hash_s[c] != hash_m.get([c],0):
                return False
                
                
        return True
    


# cheating:
# class Solution:
#     def isAnagram(self, s: str, t: str) -> bool:
#         return Counter(s) == Counter(t)